sniper/src/controller.rs

```diff
@@ -51,31 +61,70 @@ pub fn update (model: &mut Sniper, msg: Message) -> Result<Option<Message>> {
         },
         Message::OpenDir(dir_path) => {
             set_current_dir(dir_path)?;
-
-            model.file_list.elems = get_files(Path::new("."))?
-                .iter()
-                .map(|path_buf| {
-                    let raw_s = path_buf.to_string_lossy().to_string();
-                    raw_s.strip_prefix("./").map(ToOwned::to_owned).unwrap_or(raw_s) // Strip leading "./", if any
-                }).collect();
+            model.file_list.elems = get_file_entries(Path::new("."))?;
+            model.file_list.clear();
```

Comment:
I shouldn't have had to put this here. There was an out-of-bounds error, where it seemed like the index of the cursor wasn't being cleared when switching directories. But the error isn't present on main.

Reply:
Nevermind - does occur on main

Reply:
This is probably fixed by `d0e6f44`